src/components/Results.tsx

```diff
@@ -131,15 +137,18 @@ export const Results: React.FC<{
   );
 
   const handleFacilitySelection = (facilityId: string) => {
-    setSelectedFacilityId(facilityId);
+    dispatch({
+      type: "SET_SELECTED_FACILITY_ID",
+      payload: facilityId
+    });
```

Comment:
because here you calling `dispatch` you have to cover this callback with `useCallback` and put `dispatch` to the dependecies